### PR TITLE
Add support for download arm macos binaries

### LINKED ISSUE #100: zsh: exec format error: ./.local/share/bob/v0.8.3/nvim-linux64/bin/nvim
Running ARM Ubuntu on Oracle cloud. Bob installed x86_64. Installed bob using cargo install --git https://github.com/MordechaiHadad/bob.git (if it matters)

╭─ubuntu on instance-20221207-0257 in ~ 23-03-07 - 17:50:13
╰─ file ./.local/share/bob/v0.8.3/nvim-linux64/bin/nvim
./.local/share/bob/v0.8.3/nvim-linux64/bin/nvim: ELF 64-bit LSB pie executable, x86-64, version 1 (SYSV), dynamically linked, interpreter /lib64/ld-linux-x86-64.so.2, BuildID[sha1]=5d575f4797452c24c5893fdb1c0317e2d8be2340, for GNU/Linux 3.2.0, with debug_info, not stripped

### PR DESCRIPTION
**Pull request by MordechaiHadad**
fix #100 and possibly #188

### CODE REVIEW

**Comment by gibfahn** [comment]
Doesn't seem to resolve the issue unfortunately:

```console
❯ $(brew --prefix)/bin/bob install nightly
Mar 25 13:38:26.144 ERROR Error: Please provide an existing neovim version

❯ git rev-parse @
b7ed90b5fd8385477770060f3589d066dadd9fa8

❯ cargo run --release -- install nightly
    Finished release [optimized] target(s) in 0.06s
     Running `target/release/bob install nightly`
Mar 25 13:38:31.576 ERROR Error: Please provide an existing neovim version
```

File: src/helpers/mod.rs [hunk 1]
```diff
@@ -20,8 +20,10 @@ pub fn get_file_type() -> &'static str {
 pub fn get_platform_name() -> &'static str {
     if cfg!(target_os = "windows") {
         "nvim-win64"
-    } else if cfg!(target_os = "macos") {
-        "nvim-macos"
+    } else if cfg!(target_os = "macos") && cfg!(target_arch = "arm64") {
```
Comment (gibfahn) [inline src/helpers/mod.rs:23]:
```suggestion
    } else if cfg!(target_os = "macos") && cfg!(target_arch = "aarch64") {
```

https://doc.rust-lang.org/reference/conditional-compilation.html#target_arch

File: src/helpers/mod.rs [hunk 2]
```diff
@@ -30,8 +32,10 @@ pub fn get_platform_name() -> &'static str {
 pub fn get_platform_name_download() -> &'static str {
     if cfg!(target_os = "windows") {
         "nvim-win64"
-    } else if cfg!(target_os = "macos") {
-        "nvim-macos"
+    } else if cfg!(target_os = "macos") && cfg!(target_arch = "arm64") {
```
Comment (mdjohns) [inline src/helpers/mod.rs:35]:
```suggestion
    } else if cfg!(target_os = "macos") && cfg!(target_arch = "aarch64") {
```

I also had to change this line for it to work for me locally. cc @gibfahn

File: src/helpers/mod.rs (hunk 1 above)
Comment (MordechaiHadad) [reply src/helpers/mod.rs:23]: merged the other suggestion lmao

**Pull request merged**